### PR TITLE
Update GOI docs on how to run the workflow

### LINKED ISSUE #247: Add docs on how to run the GOI workflow
**Please provide some background on the proposed additions or changes.**

Once we have initiated a Snakefile specific to the genes of interest (GOI) module, we will want to update the `README.md` file with detailed information related to how to run the workflow.

**What are the changes that you are proposing?**

Update the `README.md` file with details around the required and GOI-specific parameters, the GOI specific config file, and how to run the workflow with and without changing the default parameters.

**What potential "gotchas" do we know of?**

No potential gotchas but we will want issue #229 to be completed before we describe how to run the workflow.

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #247

**What is the purpose of these changes?**

Now that we have the `goi.snakefile` merged, we want to update the `optional-goi-analysis/README.md` file to describe how to run the genes of interest workflow/snakefile.

**What changes did you make?**

This PR adds a section to the GOI `README` named `Running the workflow` that outlines the following:
- how to run the `goi.snakefile`
- the project specific parameters that are relevant in the `config/config.yaml` file
- the GOI specific parameters that are found in the `config/goi_config.yaml` file

This PR also updates the file structure section to more accurately represent the how the output files are stored.

**Any comments, concerns, or questions important for reviewers**

- Does this PR seem to adequately address #247? 
- Are there any additional details that should be added here?

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

**Review by allyhawkins — COMMENTED** [review]
A few overall comments: 

- Don't forget to update the TOC 
- I think we want to actually re-organize this... into the following sections: 
  - Analysis overview 
  - Running the workflow - here I would move up the paragraph you have about the `goi.snakefile` and then include a command to run the workflow. I would also have a paragraph after the command stating to replace the paths with paths to your results directory and project metadata. 
  - Expected input 
  - Parameters and config file - this is where you would talk about what parameters need to be altered and parameters that are specific to GOI 
  - expected output 

The goal of this reorganization is to get the actual instructions and minimal information needed to run the workflow towards the top of the readme, then to break down each of the items that are needed to create the command throughout the rest of the readme. This should mostly involve splitting the text you are adding here so that there is one overall running the workflow section which introduces the snakefile and command and then one sub section with the parameters and config file. I would also be sure to link to different sections of the readme in that main running the workflow part.

File: optional-goi-analysis/README.md [hunk 1]
```diff
@@ -33,6 +37,27 @@ R 4.2 is required for running our pipeline, along with Bioconductor 3.15.
...
+  --use-conda \
+  --config results_dir="<RELATIVE PATH TO RESULTS DIRECTORY>" \
+  project_metadata="<RELATIVE PATH TO YOUR PROJECT METADATA TSV>"
+```
+
+**You will want to replace the paths for both `results_dir` and `project_metadata` to successfully run the workflow.** 
+Where `results_dir` is the relative path to the results directory where all results from running the workflow will be stored and `project_metadata` is the relative path to the TSV file containing the relevant information about your input files.
+See more information on project metadata in the [expected input section](#expected-input) below.
+
+**Note:**  If you did not install dependencies [with conda via snakemake](../README.md#snakemakeconda-installation), you will need to remove the `--use-conda` flag.
```
Comment (allyhawkins) [inline optional-goi-analysis/README.md:58]:
```suggestion
**Note:**  If you installed dependencies [with conda via snakemake](../README.md#snakemakeconda-installation), you will need to remove the `--use-conda` flag.
```
I think this should be changed?

File: optional-goi-analysis/README.md [hunk 2]
```diff
@@ -43,20 +68,73 @@ This `SingleCellExperiment` object must contain a log-normalized counts matrix i
 3. The genes of interest list, stored as a tab-separated value (TSV) file.
 This file should contain at least one column named `gene_id` with the relevant gene identifiers, and can optionally contain an additional column named `gene_set` that denotes the gene set that each gene identifier belongs to (see an example of this genes of interest file [here](../example-data/goi-lists/sample01_goi_list.tsv)).
 
+## Parameters and config file
+
+As in the main core workflow, we have provided a [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config/config.yaml` which sets the defaults for project-specific parameters needed to run the genes of interest workflow.
+
+### Project-specific parameters
+
+There are a set of parameters included in the `config/config.yaml` file that will need to be specified when running the workflow.
```
Comment (allyhawkins) [inline optional-goi-analysis/README.md:77]:
```suggestion
There are a set of parameters included in the `config/config.yaml` file that will always need to be specified when running the workflow.
```

**Review by allyhawkins — COMMENTED** [review]
Thanks for doing this reorganization! I like it a lot. A few more comments, mostly about my opinions on what I think should really be considered an optional parameter. I think we also are going to have to think about how to discuss mapping options in a different way, but I want to do that in a separate PR.

File: optional-goi-analysis/README.md (hunk 1 above)
Comment (cbethell) [reply optional-goi-analysis/README.md:58]: Do we want to remove the `--use-conda` flag if the dependencies were installed with conda? I didn't think so but if I am wrong here we will want to change this note multiple places.

**Comment by cbethell** [comment]
Thanks for the review @allyhawkins -- I've restructured and clarified things as you suggested, with exception of one comment which I responded to above https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/281#discussion_r1091170851. Otherwise let me know what you think of things now!

File: optional-goi-analysis/README.md [hunk 3]
```diff
@@ -43,20 +68,73 @@ This `SingleCellExperiment` object must contain a log-normalized counts matrix i
...
+These include the following parameters:
+
+| Parameter        | Description |
+|------------------|-------------|
+| `results_dir` | relative path to the directory where output files will be stored (use the same `results_dir` used in the prerequisite core workflow) |
+| `project_metadata` | relative path to your specific project metadata TSV file (use the same `project_metadata` used in the prerequisite core workflow) |
+| `goi_list` | the file path to a tsv file containing the list of genes that are of interest; the default file path is `example-data/goi-lists/example_goi_list.tsv` |
+| `provided_identifier` | the type of gene identifiers used to populate the genes of interest list; example values that can implemented here include `"ENSEMBL"`, `"ENTREZID"`, `"SYMBOL"` -- where `"SYMBOL"` is the default (see more keytypes [here](https://jorainer.github.io/ensembldb/reference/EnsDb-AnnotationDbi.html)) |
+
+|[View Config File](../config/config.yaml)|
```
Comment (allyhawkins) [inline optional-goi-analysis/README.md:88]: I think we can remove this from here. They probably don't need to use this config file when what they really need for this workflow is the goi one which you have linked below.

File: optional-goi-analysis/README.md [hunk 4]
```diff
@@ -43,20 +68,73 @@ This `SingleCellExperiment` object must contain a log-normalized counts matrix i
...
+|---|
+
+The above parameters can be modified at the command line by using the [`--config` flag](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
+You must also specify the number of CPU cores for snakemake to use by using the [`--cores` flag](https://snakemake.readthedocs.io/en/stable/tutorial/advanced.html?highlight=cores#step-1-specifying-the-number-of-used-threads).
+If `--cores` is given without a number, all available cores are used to run the workflow.
+If you installed dependencies for the workflow [with conda via snakemake](../README.md#snakemakeconda-installation), you will need to provide the `--use_conda` flag as well.
+
+The below code is an example of running the genes of interest workflow using the project-specific parameters.
```
Comment (allyhawkins) [inline optional-goi-analysis/README.md:96]:
```suggestion
The below code is an example of running the genes of interest workflow using the required parameters.
```

File: optional-goi-analysis/README.md [hunk 5]
```diff
@@ -43,20 +68,73 @@ This `SingleCellExperiment` object must contain a log-normalized counts matrix i
...
+
+```
+snakemake --snakefile goi.snakefile \ 
+  --cores 2 \
+  --use-conda \
+  --config results_dir="<RELATIVE PATH TO RESULTS DIRECTORY>" \
+  project_metadata="<RELATIVE PATH TO YOUR PROJECT METADATA TSV>"
```
Comment (allyhawkins) [inline optional-goi-analysis/README.md:103]: You need to add in the goi list and identifier here.

**Review by allyhawkins — COMMENTED** [review]
A few more small comments that need to be addressed and then it should be good.

**Comment by cbethell** [comment]
I've implemented those last set of suggested changes 💯 This is ready for another look when you are @allyhawkins!

**Review by allyhawkins — APPROVED** [review]
LGTM 🚀

**Pull request merged**